src/text-editor-presenter.coffee

```diff
@@ -994,6 +1012,20 @@ class TextEditorPresenter
 
       regions
 
+  setOverlayDimensions: (decorationId, itemWidth, itemHeight, contentMargin) ->
```

Comment:
For clarity and consistency, the state object should only be used to store information that is consumed by the view. Since this information came from the view, it doesn't seem like it should be placed here. It should be just one more object for maintaining these dimensions, right? That means state update methods are free to blow away and reconstruct state without worrying about destroying data that's required to build the state. That might not happen here, but keeping a separation as a general principle means any future changes are free to reason about the state as a write-only artifact.